File: src/template.go
package main

import (
	"bytes"
	"encoding/json"
	"strings"
	"text/template"
)

type TemplateData struct {
	Env         map[string]string `json:"env"`
	Json        string            `json:"json"`
	EscapedJson string            `json:"escapedJson"`
}

func templateRoute(name string, content []byte) ([]byte, error) {
	tmpl, err := template.New(name).Parse(string(content))
	if err != nil {
		return nil, err
	}

	jsonString, err := json.Marshal(appEnv)
	if err != nil {
		return nil, err
	}

	buffer := bufferPool.Get().(*bytes.Buffer)
	defer func() {
		buffer.Reset()
		bufferPool.Put(buffer)
	}()

	err = tmpl.Execute(buffer, &TemplateData{
		Env:         appEnv,
		Json:        string(jsonString),
		EscapedJson: strings.Replace(string(jsonString), "\"", "\\\"", -1),
	})
	if err != nil {
		return nil, err
	}

	result := make([]byte, buffer.Len())
	copy(result, buffer.Bytes())
	return result, nil
}

File: src/template.rs
use anyhow::Result;
use handlebars::Handlebars;
use serde_json::json;
use std::collections::HashMap;
use std::env;

pub fn render_template(content: &str, config_prefix: &str) -> Result<String> {
    let handlebars = Handlebars::new();
    
    // Collect environment variables with the specified prefix
    let mut env_vars = HashMap::new();
    for (key, value) in env::vars() {
        if key.starts_with(config_prefix) {
            let trimmed_key = key.strip_prefix(config_prefix).unwrap();
            env_vars.insert(trimmed_key.to_string(), value);
        }
    }
    
    let json_string = serde_json::to_string(&env_vars)?;
    let escaped_json = json_string.replace('"', "\\\"");
    
    let data = json!({
        "env": env_vars,
        "json": json_string,
        "escapedJson": escaped_json
    });
    
    handlebars.render_template(content, &data)
        .map_err(|e| anyhow::anyhow!("Template rendering error: {}", e))
}